docs/drafts/architecture.md

```diff
@@ -0,0 +1,74 @@
+
+## OLM V1 Architecture
+
+This document talks about the V1 architecture. OLM V1 consists of many components, however majority of the code is divided in to two components
+
+* [operator-controller](https://github.com/operator-framework/operator-controller)
+* [catalogD](https://github.com/operator-framework/catalogd)
+
+The below diagram and text explains the sub-components within each code repository and how they related to each-other.
+
+### Operator-controller:
+
+operator-controller is the central component of OLM v1, that consumes all of the components below to extend Kubernetes to allows users to install, and manage the lifecycle of other extensions
+
+* Operator-controller queries the catalogd (catalogd HTTP Server) to get catalog information.  Once received the catalog information is saved  to catalog-cache. The cache will be updated automatically if a Catalog is noticed to have a different resolved image reference.
+* Resolver in operator-controller helps the extension controller to filter the bundle reference after applying the user restrictions (e.g. name, priority etc) and returns the bundle reference to the extension controller.
+* Cluster extension controller reaches out to the registry to download the bundle container images, saves it to the bundle cache,  unpacks it and applies the bundle manifests to the cluster.  It is also Responsible for figuring out which bundle to upgrade.
+* Bundle cache returns the cache for the bundle. If a cache does not already exist, a new one will be created.
```

Comment:
Feels odd that we are talking about the components by name before a reader sees the diagram

Reply:
Fixed it in https://github.com/operator-framework/operator-controller/pull/1301/commits/67c1540f1ed18d59b3dbc31f0fb59ff2a8a95eb4